specs/Write-Log.spec.md

```diff
@@ -0,0 +1,43 @@
+# Write-Log
+
+## Problem statement
+
+Currently, `Write-Host` Powershell cmdlet is used all over test code.
+This is done because previously used `Write-Output` messes with return values of 
+nested functions.
+
+However, the side effect of using `Write-Host` is that we have no control over
+where it prints. It always prints to console - but it interferes with nice
+formating of e.g. Pester.
+
+Also, default rules of PSScriptAnalyzer recommend against using `Write-Host`
+(https://github.com/PowerShell/PSScriptAnalyzer/blob/3dc995d7a9eaf259077c469ad8468aefb8eed133/RuleDocumentation/AvoidUsingWriteHost.md)
+
+Recommendation is to use `Write-Output` or `Write-Verbose`.
+
+`Write-Output` is problematic, for the reasons stated above.
+
+`Write-Verbose` is nice, but it's hard to redirect its output to e.g. a file.
```

Comment:
It seems that `Write-Verbose` would be perfect here. Why is it hard to redirect it to a file?

Reply:
I don't think we want clients of `Write-Verbose` to specify how and where the log file should be stored. We probably want to control it globally, e.g. we don't want users to write code like:

`Write-Verbose "asdf" -Level Warn | Out-File my-logfile.txt`

but

`Write-Log -Warn "asdf"`

And somewhere in `Init.ps1` we could have `Set-LogOutputFile asdf`

Reply:
Also, Write-Verbose uses some pipes which might be problematic with `Invoke-NativeCommand`? @michal-clapinski

Reply:
wrt. `Invoke-NativeCommand`, if it will be properly split into two variants (capturing and non capturing), then it will be simple to adapt to whichever log approach we choose.